crates/ruff/src/rules/flake8_return/helpers.rs

```diff
@@ -16,3 +21,29 @@ pub fn result_exists(returns: &[(&Stmt, Option<&Expr>)]) -> bool {
         .unwrap_or(false)
     })
 }
+
+/// Check if the given piece code is composed of exclusively comments
+pub fn code_is_only_comments(code: &str) -> bool {
+    let dedented = textwrap::dedent(code);
+    let code_tokens = lexer::make_tokenizer(&dedented).flatten();
+    for (_, tok, _) in code_tokens {
+        if !matches!(
+            tok,
+            Tok::Comment(..)
+                | Tok::String { .. }
+                | Tok::Pass
+                | Tok::Newline
+                | Tok::NonLogicalNewline
+        ) {
+            return false;
+        }
+    }
+    true
+}
+
+/// Extract the indentation from a given line
+pub fn extract_indentation(line: &str, stylist: &Stylist) -> String {
+    line.chars()
+        .take_while(|c| stylist.indentation().contains(*c))
+        .collect()
+}
```

Comment:
Could we perhaps use the `indentation` helper in `crates/ruff/src/ast/whitespace.rs`?

Reply:
Maybe? But I would guess it would be slower because I would have to create a bunch of useless intermediate objects because of the way that the function takes in input as a `Located<T>`, and I mainly just have a `&str`

Reply:
It would also be kind of inconvenient. I might have to iterate through it in a sliding window fashion manually (it would also be more performant).

Reply:
To clarify, the function you are referring to would be inconvenient either way, and rewriting without `TupleWindows` would be faster either way.